version: '3.8'
services:
  mlflow:
    container_name: mlflow
    build:
      context: .
      dockerfile: dockerfile.mlflow
    ports:
      - "5000:5000"
    volumes:
      - ${PWD}:/home/mlflow/
      - ${PWD}/credentials/:/credentials/
    environment:
      - GOOGLE_APPLICATION_CREDENTIALS=/credentials/service-account.json
      - GOOGLE_CLOUD_PROJECT=logical-sled-430104-t6
    networks:
      - app_network

  mage:
    image: mageai/mageai:latest
    container_name: mage
    command: /app/run_app.sh mage start mage
    ports:
      - "6789:6789"
    volumes:
      - ./mage:/home/src/mage
      - ./mage_data:/home/src/mage_data
      - ./deployment:/home/src/deployment
      - ./credentials/:/home/src/credentials/
    environment:
      - USER_CODE_PATH=/home/src/mage
      - MLFLOW_TRACKING_URI=http://mlflow:5000
      - GOOGLE_APPLICATION_CREDENTIALS=credentials/service-account.json
      - GOOGLE_CLOUD_PROJECT=logical-sled-430104-t6
      - DATA_SET_BUCKET=credit-card-fraud-prediction-tes
      - MODEL_REGISTRY_BUCKET=mlops-test-buckt
      - BIGQUERY_DATASET=model_monitoring
      - BIGQUERY_TABLE=model_predictions
    networks:
      - app_network
    depends_on:
      - mlflow


networks:
  app_network:
    driver: bridge
